pkg/query/storeset.go

```diff
@@ -311,6 +333,43 @@ func externalLabelsFromStore(st *storeRef) string {
 	return tsdbLabels.String()
 }
 
+func (s *StoreSet) updateStoreStatus(store *storeRef, err error) {
+	now := time.Now()
+
+	s.storesStatusesMtx.Lock()
+	defer s.storesStatusesMtx.Unlock()
+
+	minTime, maxTime := int64(-1), int64(-1)
+	if err == nil {
```

Comment:
You have this case https://github.com/improbable-eng/thanos/blob/93bae5cd40198f6d8fef00f3602b4c180b6042b8/pkg/query/storeset.go#L294 where there's no store instance yet that would cause those to be `nil` for example ?